api/src/context.rs

```diff
@@ -136,6 +130,109 @@ impl Context {
         self.db.get_block_timestamp(version)
     }
 
+    /// Retrieves information about a block
+    pub fn get_block_info(&self, version: u64, ledger_version: u64) -> Result<Option<BlockInfo>> {
+        // We scan the DB to get the block boundaries
+        let (start, end) = match self.db.get_block_boundaries(version) {
+            Ok(inner) => inner,
+            Err(error) => {
+                debug!("Failed to get block boundaries {}", error);
+                // None means we can't find the block
+                return Ok(None);
+            }
+        };
+
+        let txn_with_proof = self
+            .db
+            .get_transaction_by_version(start, ledger_version, false)?;
+
+        // Retrieve block timestamp and hash
+        let timestamp;
+        let block_hash;
+        use aptos_types::transaction::Transaction::*;
+        match &txn_with_proof.transaction {
+            GenesisTransaction(_) => {
+                timestamp = 0;
+                block_hash = txn_with_proof.proof.transaction_info.transaction_hash();
+            }
+            BlockMetadata(inner) => {
+                timestamp = inner.timestamp_usecs();
+                block_hash = inner.id();
+            }
+            _ => {
+                return Err(anyhow!(
+                    "Failed to retrieve BlockMetadata or Genesis transaction"
+                ));
+            }
+        }
+
+        // If timestamp is 0, it's the genesis transaction, and we can stop now
+        if timestamp == 0 {
+            return Ok(Some(BlockInfo {
+                block_height: 0,
+                start_version: start,
+                end_version: end,
+                block_hash: block_hash.into(),
+                block_timestamp: timestamp,
+                num_transactions: end.saturating_sub(start).saturating_add(1) as u16,
+            }));
+        }
+
+        // Retrieve block height from the transaction outputs
+        let height_id = ident_str!("height");
+        let block_metadata_type = move_deps::move_core_types::language_storage::StructTag {
+            address: AccountAddress::ONE,
+            module: ident_str!("Block").into(),
+            name: ident_str!("BlockMetadata").into(),
+            type_params: vec![],
+        };
+
+        let resolver = self.move_resolver()?;
+        let converter = resolver.as_converter();
+        let txn = self.get_transaction_by_version(start, ledger_version)?;
+
+        // Parse the resources and find the block metadata resource update
+        let maybe_block_height = txn.changes.iter().find_map(|(key, op)| {
+            if let StateKey::AccessPath(path) = key {
+                if let Path::Resource(typ) = path.get_path() {
+                    // If it's block metadata, we can convert it to get the block height
+                    // And it must be the root address
+                    if path.address == aptos_root_address() && typ == block_metadata_type {
```

Comment:
can't you just make a `StateKey` that represents `root_addr::[0x1::Block::BlockMetadta]` (and make it static), and search the change set by that?

Reply:
I'm a noob, and kinda been just doing it all on my own.  If that works, then I'll do that instead!